### PR TITLE
Add main.yml github actions file

### PR DESCRIPTION
**Pull request by JackWilb**
Adds github actions functionality

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: CI
+
+on: 
+  pull_request:
+    branches:
+    - master
+  
+  push:
+    branches:
+    - master
```
Comment (waxlamp) [inline .github/workflows/main.yml]: Indent 2 spaces

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.7'
+        architecture: 'x64'
+    - run: pip install pipenv
+    - run: pipenv install --dev --deploy
+    - run: pipenv run lint
+    - run: pipenv run format
+    - run: pipenv run typecheck
+    - run: pipenv run test
```
Comment (waxlamp) [inline .github/workflows/main.yml]: Indent 2 spaces

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+
+    - uses: actions/setup-node@v1
+      with:
+        node-version: '10.x'
+    - run: cd client && yarn install
+    - run: cd client && yarn lint
```
Comment (waxlamp) [inline .github/workflows/main.yml]: Indent 2 spaces

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        python-version: '3.7'
+        architecture: 'x64'
+    - run: pip install pipenv
+    - run: pipenv install --dev --deploy
+    - run: pipenv run lint
+    - run: pipenv run format
+    - run: pipenv run typecheck
+    - run: pipenv run test
+
+
```
Comment (waxlamp) [inline .github/workflows/main.yml]: Remove one blank line

**Comment by JackWilb** [comment]
> I don't really love the idea of the single job with "build" and "test" shoved together. Can we separate out the build phase from the various types of testing we're doing? We had a similar setup in Circle, but there at least each testing phase had its own clear title.
> 
> UPDATE: I disconnected CircleCI from this org, so I think that means we won't get any more failing results from CircleCI on subsequent pushes (or at least, for new branches after this is merged).

I had to disable it in the branch protections too. Should be good to merge now

**Pull request merged**